### PR TITLE
Complete User story 10: new user registration

### LINKED ISSUE #10: User Story 10, User Registration
As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in

### PR DESCRIPTION
**Pull request by KirbyDD**
## Description
Resolves #10 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
```
Finished in 3.25 seconds (files took 4.53 seconds to load)
104 examples, 0 failures

Coverage report generated for RSpec to /Users/tronkat/Turing/BackEnd/Mod2/projects/monster_shop_part_1/coverage. 1252 / 1254 LOC (99.84%) covered.
```

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    user = User.new(user_params)
+
+    if user.save
+      session[:user_id] = user.id
+      flash[:notice] = "Avast! Ye be registered and logged in!"
+      redirect_to '/profile'
+    else
+      flash[:notice] = "Scupper That! Ye did not complete all fields!"
+      render "new"
+    end
```
Comment (DanielEFrampton) [inline app/controllers/users_controller.rb:15]: Need to refactor this later to shorten length of method to 5 lines

**Review by philjdelong — COMMENTED** [review]
Looks good!

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<h2>Ahoy, me hearties! Be ye welcome to Treasure Trove.</h2>
+
```
Comment (DanielEFrampton) [inline app/views/home/index.html.erb:2]: Something Demarcus and I removed from our first draft of the home page was a conditional message encouraging the user to login or register if not logged in. We can return to that later after we resolve the various sad paths.

**Review by DanielEFrampton — APPROVED** [review]
Great work everybody.

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class HomeController < ApplicationController
```
Comment (mel-rob) [inline app/controllers/home_controller.rb:1]: Can we change this to WelcomeController?

**Review by mel-rob — DISMISSED** [review]

**Review by mel-rob — APPROVED** [review]

**Review by philjdelong — APPROVED** [review]

**Pull request merged**